### PR TITLE
detect: xor transform

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3285

Describe changes:
- Adds xor as transform

suricata-verify-pr: 574
https://github.com/OISF/suricata-verify/pull/574

Replaces #6533 with commit wording changes

### CODE REVIEW

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | test | baseline | % |
|--|--|--|--| 
| | tlpr1_stats_chk |
| .flow.wrk.flows_evicted | 5306518 | 4693221 | 113.07% |
| .flow.mgr.rows_maxlen | 115 | 539 | 21.34% | 

Pipeline 4677

File: src/detect-transform-xor.c [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    // Create pxd from optstr
+    DetectTransformXorData *pxd = SCCalloc(1, sizeof(*pxd));
+    if (pxd == NULL) {
+        SCLogError(SC_ERR_MEM_ALLOC, "memory allocation failed");
+        SCReturnInt(-1);
+    }
+
+    size_t keylen = strlen(optstr);
+    if (keylen % 2 == 1) {
+        SCLogError(SC_ERR_INVALID_SIGNATURE, "XOR transform key must be an even number");
```
Comment (victorjulien) [inline src/detect-transform-xor.c:90]: should the key or the keys length be an even number?

Comment (catenacyber) [reply src/detect-transform-xor.c:90]: key's length indeed

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/6579

**Pull request closed without merging**